﻿<!doctype html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>Co模块</title>
</head>
<body>
<script>
    function delay(isOut) {
        return new Promise((resolve, reject) => {
            setTimeout(function () {
                reject(new Error("in error"))
            }, 1000)
        })
    }

    const byPromise = function (fn) {
        return function (...args) {
            return fn.call(this, ...args)
        }
    }

    async function Foo() {
        try {
            await byPromise(delay)(true).catch(err => {
                console.log(err)
            })
        } catch (err) {
            console.log(err)
        }
    }
    Foo().catch(err => {
        console.log(err)
    })
    // Error: out error  白字错误
    // Uncaught Error: in error     红字错误

</script>
</body>
</html>